ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -1307,6 +1292,83 @@ module RotateMachinePassword = struct
   let stop_rotate () = Scheduler.remove_from_queue task_name
 end
 
+module type LocalHostTag = sig
+  val local_ip : string
+end
+
+module HostsConfTagIPv4 : LocalHostTag = struct let local_ip = "127.0.0.1" end
+
+module HostsConfTagIPv6 : LocalHostTag = struct let local_ip = "::1" end
+
+module type HostsConf = sig
+  (* add the domain info into conf*)
+  val join : name:string -> domain:string -> lines:string list -> string list
+
+  (* remove the domain info from conf*)
+  val leave : name:string -> domain:string -> lines:string list -> string list
+end
+
+module HostsConfFunc (T : LocalHostTag) : HostsConf = struct
+  let sep = ' '
+
+  let sep_str = String.make 1 sep
+
+  type t = Add | Remove
+
+  let interest line = String.starts_with ~prefix:T.local_ip line
+
+  let handle op name domain line =
```

Comment:
I am not sure this is the best way to handle a line in /etc/hosts. A line could be split into a list of words (like is done here) and then pattern matched to have more control over what we are doing:

```
match word lines with
| "#" :: _ -> (* comment *)
| [] -> (* empty *) 
| ip :: name :: aliases when name = ...  || when name = fqdn -> 
...
```
 I would have more trust that the code above handles the different cases correctly.

Reply:
The drawback here is we need to split and concat every line in the config, even the line is out of interest.

And we need to iterate every alias to ensure the netbios name is added/removed.